src/platform.rs

```diff
@@ -0,0 +1,98 @@
+use std::collections::HashMap;
+use std::env::VarError;
+use std::ffi::{OsStr, OsString};
+use std::fs;
+use std::io;
+use std::path::Path;
+
+/// Represents a Cloud Native Buildpack platform.
+///
+/// Most buildpacks target a generic platform and this library provides a [`GenericPlatform`] for that
+/// use-case. Buildpack authors usually do not need to implement this trait. See
+/// [detection](https://github.com/buildpacks/spec/blob/main/buildpack.md#detection) and
+/// [build](https://github.com/buildpacks/spec/blob/main/buildpack.md#build) in the buildpack
+/// specification for details.
+pub trait Platform
+where
+    Self: Sized,
+{
+    /// Retrieve a [`PlatformEnv`] reference for convenient access to environment variables which
+    /// all platforms have to provide.
+    fn env(&self) -> &PlatformEnv;
+
+    /// Initializes the platform from the given platform directory.
+    ///
+    /// # Examples
+    /// ```no_run
+    ///use libcnb::platform::Platform;
+    ///use libcnb::platform::GenericPlatform;
+    ///let platform = GenericPlatform::from_path("/platform").unwrap();
+    /// ```
+    fn from_path(platform_dir: impl AsRef<Path>) -> io::Result<Self>;
+}
+
+/// A generic platform that only provides access to environment variables.
+pub struct GenericPlatform {
+    env: PlatformEnv,
+}
+
+impl Platform for GenericPlatform {
+    fn env(&self) -> &PlatformEnv {
+        &self.env
+    }
+
+    fn from_path(platform_dir: impl AsRef<Path>) -> io::Result<Self> {
+        Ok(GenericPlatform {
+            env: PlatformEnv::from_path(platform_dir)?,
+        })
+    }
+}
+
+/// Provides access to platform environment variables.
+pub struct PlatformEnv {
+    vars: HashMap<OsString, String>,
+}
+
+impl PlatformEnv {
```

Comment:
should we implement a `fn vars` or do you think people won't want an iterator?

Reply:
I think I stopped there since I couldn't properly emulate the `std::env` API and we discussed we wanted that. I'll leave this unimplemented for now.